File: programming/DS /cpp/searching/linearSearch.cpp
#include<iostream>
#include<vector>
using namespace std;
void linearSearch(int *arr,int key,int n){
    for(int i=0;i<10;i++){
        if(arr[i]==key){
            cout<<"key found at index"<<i+1<<endl;
            return;
        }
    }cout<<"key not found!"<<endl;
}
int main(){
    int arr[10];

    for(int i=0;i<10;i++){
        arr[i]=i+1;
    }
    linearSearch(arr,20,10);

}

File: programming/javaClasses/Arrays/linearSearch.java
import java.util.*;
class linearSearch{
    public static int linearSearch(int arr[],int key){
        for(int i=0;i<arr.length;i++){
            if(arr[i]==key) return 1;

        }return 0;             
    }
    public static void main(String args[]){
        Scanner sc=new Scanner(System.in);
        int arr[]=new int[5];
        int key;
        for(int i=0;i<5;i++){
            arr[i]=sc.nextInt();
        }
        System.out.println("Enter the key you want to search:");
        key=sc.nextInt();
        System.out.println(linearSearch(arr,key));
    }
}